### PR TITLE
feat(core): set doc mode and primary doc mode separately

### PR DESCRIPTION
**Pull request by JimmFly**
https://github.com/user-attachments/assets/98c282f2-4c53-475f-bf10-936a626c2630

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#8359** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8359?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/blocksuite/block-suite-header/menu/index.tsx [hunk 1]
```diff
@@ -148,16 +149,22 @@ export const PageHeaderMenuButton = ({
   }, [rename]);
 
   const handleSwitchMode = useCallback(() => {
-    editorService.editor.toggleMode();
+    const mode = primaryMode === 'page' ? 'edgeless' : 'page';
+    editorService.editor.doc.setPrimaryMode(mode);
```
Comment (forehalo) [inline packages/frontend/core/src/components/blocksuite/block-suite-header/menu/index.tsx:153]: updating primary mode only happens in page property settings

Comment (JimmFly) [reply packages/frontend/core/src/components/blocksuite/block-suite-header/menu/index.tsx:153]: But changing the primary mode in the menu is by design

File: packages/frontend/i18n/src/resources/en.json [hunk 2]
```diff
@@ -91,7 +91,7 @@
   "Storage and Export": "Storage and export",
   "Successfully deleted": "Successfully deleted",
   "Successfully joined!": "Successfully joined!",
-  "Switch": "Switch",
+  "Switch": "Switch view",
```
Comment (EYHN) [inline packages/frontend/i18n/src/resources/en.json]: create a new key?

**Review by EYHN — APPROVED** [review]

File: packages/frontend/i18n/src/resources/en.json (hunk 2 above)
Comment (JimmFly) [reply packages/frontend/i18n/src/resources/en.json]: That's fine too.

**Pull request merged**